config.go

```diff
@@ -706,6 +707,41 @@ type DBSettings struct {
 	DB                   string       `yaml:"db"`
 	WriteConcernSettings WriteConcern `yaml:"write_concern"`
 	ReadConcernSettings  ReadConcern  `yaml:"read_concern"`
+	AuthFile             string       `yaml:"auth_file"`
+}
+
+func (dbs *DBSettings) HasAuth() bool {
+	return dbs.AuthFile != ""
+}
+
+type dbCreds struct {
+	DBUser string `yaml:"mdb_database_username"`
+	DBPwd  string `yaml:"mdb_database_password"`
+}
+
+func (dbs *DBSettings) GetAuth() (string, string, error) {
+	return GetAuthFromYAML(dbs.AuthFile)
+}
+
+func GetAuthFromYAML(authFile string) (string, string, error) {
+	var user, pwd string
+	creds := &dbCreds{}
+
+	file, err := os.Open(authFile)
+	if err != nil {
+		return user, pwd, err
```

Comment:
I think it's better to be explicit here so that a later reader doesn't have to read the beginning of the function to realize `user` and `pwd` are empty. Instead,
`return "", "", err`